include/rsl/monad.hpp

```diff
@@ -121,6 +122,23 @@ template <typename T, typename E>
     return exp.has_value();
 }
 
+/**
+ * @brief Steal value out of a non-null optional
+ *
+ * @param[in] optional A non-null optional
+ *
+ * @tparam T Optional value type
+ *
+ * @return Value inside the optional
+ */
+template <typename T>
+[[nodiscard]] constexpr auto unwrap(std::optional<T>& optional) {
+    assert(optional.has_value());
+    auto value = std::move(*optional);
```

Comment:
If we mark this as `const`, the "move-only" test fails which I thought was neat. We should try harder to test our function templates with move-only types like `std::future` or `std::unique_ptr` since they can catch subtle issues with the over-use of `const`.